tests/test_stateful_browser.py

```diff
@@ -426,6 +426,30 @@ def test_select_form_tag_object():
             browser.select_form(soup.find("p"))
 
 
+def test_select_form_associated_elements():
+    """Test associated elements outside the form tag"""
+    forms = """<form id="a"><input><textarea></form><input form="a">
+               <textarea form="a"/><input form="b">
+               <form id="ab" action="/test.php"><input></form>
+               <textarea form="ab"></textarea>
+            """
+    with mechanicalsoup.StatefulBrowser() as browser:
+        browser.open_fake_page(forms)
+        elements_form_a = set([
+            "<input/>", "<textarea></textarea>",
+            '<input form="a"/>', '<textarea form="a"></textarea>'])
+        elements_form_ab = set(["<input/>", '<textarea form="ab"></textarea>'])
+        form_by_str = browser.select_form("#a")
+        form_by_tag = browser.select_form(browser.page.find("form", id='a'))
+        form_by_css = browser.select_form("form[action$='.php']")
+        assert set(list(map(str, form_by_str.form.find_all((
```

Comment:
I understand, maybe a list comprehension is easier to read, I use it sometimes when I need to filter data, like `[ x for x in foo if x == ... ]`, otherwise I use map, but I can change it anyway.